src/Renderer/Label2DRenderer.js

```diff
@@ -155,6 +190,8 @@ class Label2DRenderer {
         // visible extent, we can filter more labels.
         } else if (object.zoom.max <= currentMaxZoom || !extent.isPointInside(object.coordinates)) {
             this.grid.hidden.push(object);
+        } else if (object.parent.level < 4 && object.parent.layer.horizonCulling && object.parent.layer.horizonCulling(object.horizonCullingPoint)) {
```

Comment:
could you explain `object.parent.level < 4`?

Reply:
Done